### PR TITLE
Fix DCHECK on secure rpc

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
If anyone call GetHandshakeContext with invalid application id
application manager should not assert this case, just return empty SSLContext

Related Issue : APPLINK-24766

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@AByzhynar @AGritsevich @dtrunov @okozlovlux @nk0leg @AGaliuzov  please review

**Comment by AGritsevich** [comment]
Reviewed.

**Comment by okozlovlux** [comment]
@LuxoftAKutsan reviewed

**Comment by AByzhynar** [comment]
Reviewed

**Comment by nk0leg** [comment]
Reviewed

File: src/components/application_manager/src/application_manager_impl.cc [hunk 1]
```diff
@@ -1262,8 +1262,10 @@ security_manager::SSLContext::HandshakeContext
 ApplicationManagerImpl::GetHandshakeContext(uint32_t key) const {
   LOG4CXX_AUTO_TRACE(logger_);
   ApplicationConstSharedPtr app = application(key);
+  if (!app) {
```
Comment (AGaliuzov) [inline src/components/application_manager/src/application_manager_impl.cc]: I would raher re-write this method as 

>security_manager::SSLContext::HandshakeContext res;
if (app) {
    res.make_context(custom_str::CustomString(app->policy_app_id()),  app->name());
}
return res;

**Comment by dtrunov** [comment]
Reviewed

**Comment by LuxoftAKutsan** [comment]
SSLContext please re review

**Comment by LuxoftAKutsan** [comment]
@AByzhynar @AGritsevich @dtrunov @okozlovlux @nk0leg @AGaliuzov please **re review**

File: src/components/security_manager/test/ssl_context_test.cc [hunk 2]
```diff
@@ -237,8 +237,8 @@ class SSLTestParam : public testing::TestWithParam<ProtocolAndCipher> {
     client_ctx = client_manager->CreateSSLContext();
 
     security_manager::SSLContext::HandshakeContext ctx;
-    ctx.make_context(custom_str::CustomString("SPT"),
-                     custom_str::CustomString("client"));
+    ctx.expected_cn = "SPT";
```
Comment (AGaliuzov) [inline src/components/security_manager/test/ssl_context_test.cc]: @LuxoftAKutsan So, use the cinstructor you have created

**Comment by dtrunov** [comment]
Reviewed

File: src/components/security_manager/test/ssl_context_test.cc [hunk 3]
```diff
@@ -158,8 +158,8 @@ class SSLTest : public testing::Test {
     client_ctx = client_manager_->CreateSSLContext();
 
     security_manager::SSLContext::HandshakeContext ctx;
-    ctx.make_context(custom_str::CustomString("SPT"),
-                     custom_str::CustomString("client"));
+    ctx.expected_cn = "SPT";
+    ctx.expected_sn = "client";
     server_ctx->SetHandshakeContext(ctx);
```
Comment (nk0leg) [inline src/components/security_manager/test/ssl_context_test.cc]: why you don't use new constructor?

**Comment by AGritsevich** [comment]
Reviewed

File: src/components/security_manager/test/ssl_context_test.cc (hunk 3 above)
Comment (LuxoftAKutsan) [inline src/components/security_manager/test/ssl_context_test.cc]: @nk0leg  in previous version it was one object< i do not want to add some unexpected regression

Comment (LuxoftAKutsan) [inline src/components/security_manager/test/ssl_context_test.cc]: @nk0leg  please see 086bdb1

File: src/components/security_manager/test/ssl_context_test.cc (hunk 2 above)
Comment (LuxoftAKutsan) [inline src/components/security_manager/test/ssl_context_test.cc]: @AGaliuzov please see 086bdb1

**Comment by nk0leg** [comment]
Reviewed

**Comment by AGaliuzov** [comment]
Reviewed

**Comment by okozlovlux** [comment]
@LuxoftAKutsan reviewed

**Comment by AByzhynar** [comment]
Re-reviewed

**Comment by LuxoftAKutsan** [comment]
@AGaliuzov  please merge

**Comment by AGaliuzov** [comment]
@LuxoftAKutsan why the PR contains changes for travic and check_style.sh?

**Comment by LuxoftAKutsan** [comment]
@Agaliuzov We do not have any specific task to modify release branch configurations.
This changes are realy useful.

**Pull request merged**